### PR TITLE
Reorganization of collections code

### PR DESCRIPTION
**Pull request by mtopolnik**
1. Fixed constructor signatures and Javadoc in HashSlotArray8byteKey.
2. Moved QuickSort classes to a fitting package (from `...memory` to `...internal.util.sort`)
3. Changes in  `Long2LongX` API: 
    1. Reimplement classes in terms of `HashSlotArray`
    2. Replace `Iterator` with `Cursor`
    3. Remove "elastic" word from name
    4. Move from EE to OSS

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/util/collection/LongSetHsa.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        this.nullValue = nullValue;
+        this.hsa = new HashSlotArray8byteKeyNoValue(nullValue, memMgr, initialCapacity, loadFactor);
+        hsa.gotoNew();
+    }
+
+    @Override
+    public boolean add(long value) {
+        assert value != nullValue : "add() called with null-sentinel value " + nullValue;
+        return hsa.ensure(value) > 0;
+
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/internal/util/collection/LongSetHsa.java]: Why the extra white space?

File: hazelcast/src/test/java/com/hazelcast/internal/util/collection/Long2LongMapHsaTest.java [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+
+    @Test
+    public void testMemoryLeak() {
+        int keyRange = 100;
+
+        for (int i = 0; i < 100000; i++) {
+            int k = random.nextInt(7);
+            switch (k) {
+                case 0:
+                    _put_(keyRange);
```
Comment (pveentjer) [inline hazelcast/src/test/java/com/hazelcast/internal/util/collection/Long2LongMapHsaTest.java]: Why these underscores?

**Comment by pveentjer** [comment]
Looks good :+1:

File: hazelcast/src/main/java/com/hazelcast/internal/util/collection/LongSetHsa.java (hunk 1 above)
Comment (mtopolnik) [inline hazelcast/src/main/java/com/hazelcast/internal/util/collection/LongSetHsa.java]: where? I don't see it

File: hazelcast/src/test/java/com/hazelcast/internal/util/collection/Long2LongMapHsaTest.java (hunk 2 above)
Comment (mtopolnik) [inline hazelcast/src/test/java/com/hazelcast/internal/util/collection/Long2LongMapHsaTest.java]: This is old code, just kept it as-is.

**Comment by jerrinot** [comment]
:+1:

**Pull request merged**